### ISSUE TITLE
isCancelled() is returning false for Android Subscription

### ISSUE THREAD

**Issue by sabinayakc** [voltrue2/in-app-purchase#328]
#### Description:
When an android subscription is cancelled, `iap.isCancelled()` returns false. Looking at the PurchaseItem response, there is no field called cancellationDate for android. Shouldn't the code below be checking for `userCancellationTimeMillis` for Android instead of cancellationDate? Same goes for isExpired() function.

_I haven't checked for IOS._

https://github.com/voltrue2/in-app-purchase/blob/e966ee1348bd4f67581779abeec59c4bbc2b2ebc/index.js#L261

An example subscriptionResponse:

{
>    service: 'google',
>    status: 0,
>    packageName: 'XXX',
>    productId: 'XXX',
>    purchaseToken: 'XXX',
>    startTimeMillis: '1600461854942',
>    expiryTimeMillis: '1600462273216',
>    autoRenewing: false,
>    priceCurrencyCode: 'USD',
>    priceAmountMicros: '9990000',
>    countryCode: 'US',
>    developerPayload: '',
>    cancelReason: 0,
>    **userCancellationTimeMillis: '1600461907018'**,
>    orderId: 'XXXXX,
>    purchaseType: 0,
>    acknowledgementState: 1,
>    kind: 'androidpublisher#subscriptionPurchase'
>  }

**Comment by KochMario** [comment]
You're absolutely right. I was wondering the same thing while validating my Android subscriptions.
Possibly the Google API changed and @voltrue2 hasn't noticed or hasn't had the time to update.

For the time being, I added an additional validator for android.
